It seems there is a real problem.


```
    DEBUG [pifpaf.drivers] executing: ['gnocchi-upgrade', '--config-file=/tmp/tmpxhl8x7yy/gnocchi.conf']
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output: Traceback (most recent call last):
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/home/pifpaf/pifpaf/.tox/py313/bin/gnocchi-upgrade", line 3, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from gnocchi.cli.manage import upgrade
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/home/pifpaf/pifpaf/.tox/py313/lib/python3.13/site-packages/gnocchi/cli/manage.py", line 27, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from gnocchi import service
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/home/pifpaf/pifpaf/.tox/py313/lib/python3.13/site-packages/gnocchi/service.py", line 26, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from gnocchi import opts
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/home/pifpaf/pifpaf/.tox/py313/lib/python3.13/site-packages/gnocchi/opts.py", line 25, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     import gnocchi.rest.http_proxy_to_wsgi
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/home/pifpaf/pifpaf/.tox/py313/lib/python3.13/site-packages/gnocchi/rest/http_proxy_to_wsgi.py", line 16, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     import webob.dec
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/usr/lib/python3/dist-packages/webob/__init__.py", line 1, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from webob.datetime_utils import (  # noqa: F401
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     ...<13 lines>...
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     )
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/usr/lib/python3/dist-packages/webob/datetime_utils.py", line 18, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from webob.compat import (
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     ...<4 lines>...
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:         )
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:   File "/usr/lib/python3/dist-packages/webob/compat.py", line 5, in <module>
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output:     from cgi import parse_header
   DEBUG [pifpaf.drivers] gnocchi-upgrade[1787] output: ModuleNotFoundError: No module named 'cgi'
```

--- a/.github/workflows/pifpaf.yaml
+++ b/.github/workflows/pifpaf.yaml
@@ -19,6 +19,7 @@ jobs:
           - py310
           - py311
           - py312
+          - py313
           - pep8
           - build
     steps:

--- a/Dockerfile
+++ b/Dockerfile
@@ -14,6 +14,7 @@ RUN apt-get -qq update -y \
           python3 python3-dev python3-pip python3-virtualenv \
           python3.10 python3.10-dev python3.10-distutils \
           python3.11 python3.11-dev \
+          python3.13 python3.13-dev \
           gcc liberasurecode-dev liberasurecode1 postgresql libpq-dev python3-rados git wget memcached \
     && rm -rf /var/lib/apt/lists/*
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -16,6 +16,7 @@ classifier =
     Programming Language :: Python :: 3.10
     Programming Language :: Python :: 3.11
     Programming Language :: Python :: 3.12
+    Programming Language :: Python :: 3.13
     Topic :: Software Development :: Testing
 
 [options]
@@ -39,6 +40,7 @@ test =
     stestr
     testtools
     mock
+    legacy-cgi; python_version>='3.13'
 gnocchi =
     uwsgi
 

